MIPs/mip-3.md

```diff
@@ -0,0 +1,184 @@
+   Currently we throw this errors:
+
+- **Undefined Parameters**: Parameters like `from`, `to`, `user_address`, etc. are essential for the `wallet_swapAsset` method to work correctly. If not provided, the `validateParams` function would throw an error saying `"${property} property of ${name} is not defined"`. (This validation will change on the future when we allow multiple swap tokens, since the architecture of this rpc method allows it.)
+
+  ```markdown
+  validateParams(from[0], [amount, token_address], 'from');
+  validateParams(to, [token_address], 'to');
+  ```
+
+- **Invalid User Address**: If a non-existent address is provided in `user_address`, then the method will throw an error 'This address does not exist'.
+
+  ```markdown
+  if (!dappConnectedAccount) {
+  throw ethErrors.rpc.invalidParams('This address does not exist');
+  }
+  ```
+
+- **Unsupported Multiple Tokens Swap**: As of now, it supports only a single token swap. In case of multiple tokens provided in the `from` object, an 'Currently we de not support multiple tokens swap' error is thrown.
+
+  ```markdown
+  if (from.length > 1) {
+  throw ethErrors.rpc.invalidParams(
+  'Currently we de not support multiple tokens swap',
+  );
+  }
+  ```
+
+- **Inactive or Unsupported Swaps**: If the swap is inactive or not possible on the current chain, an alert with the message 'Swap is not active or not possible on this chain' will be triggered.
+
+  ```markdown
+  const isSwappable = isSwapsAllowed(chainId) && swapsIsLive;
+  if (!isSwappable) {
+  Alert.alert('Swap is not active or not possible on this chain');
+  return;
+  }
+  ```
+
+5. Security: Dapp Developers should ensure that the token addresses and chain IDs provided in the method call are valid and secure. They should also inform users about the potential risks and considerations when performing token swaps.
+   On MetaMask platfors we have warnings in place that say to the user to be carefull when a token is not that trustworthy. We check if the number of occurrences is more than one, If not we show an alert the we obligate the user to ready and press a button to be able to swap.
+
+6. Testing: Before deploying the `wallet_swapAsset` method in a live environment, MetaMask engineers should thoroughly test it to ensure it works correctly and handles errors appropriately. This includes testing with different token types, amounts, and network conditions.
+
+7. Interoperability: The `wallet_swapAsset` method provides a standardized way for dApps to request token swaps from wallets. This can improve interoperability between different dApps and wallets, as they can all use the same method for token swaps.
+
+## User Experience Considerations
+
+Users will now be able to swap tokens without leaving dapps.
+The `wallet_swapAsset` method might be new to some users, providing educational resources can help them understand how it works and how to use it.
```

Comment:
```suggestion
Since `wallet_swapAsset` method will be new to some users, providing educational resources may help in understand how the method works as well as how to use it.
```